src/snowflake/connector/network.py

```diff
@@ -170,6 +172,32 @@ def is_retryable_http_code(code: int) -> bool:
     )
 
 
+class ProxySupportAdapter(HTTPAdapter):
+    """This Adapter creates proper headers for Proxy CONNECT messages."""
+
+    def get_connection(self, url, proxies=None):
+        proxy = select_proxy(url, proxies)
+
+        if proxy:
+            proxy = prepend_scheme_if_needed(proxy, 'http')
+            proxy_url = parse_url(proxy)
+            if not proxy_url.host:
+                raise InvalidProxyURL("Please check proxy URL. It is malformed"
+                                      " and could be missing the host.")
+            proxy_manager = self.proxy_manager_for(proxy)
+            parsed_url = urlparse(url)
```

Comment:
Could this line be moved out of the if statement, please? It seems to be used by both the if and else clause

Reply:
Marking this as resolved.